message_definitions/v1.0/ardupilotmega.xml

```diff
@@ -1708,5 +1708,19 @@
       <field type="float" name="min_distance" units="m">Minimum distance the sensor can measure.</field>
       <field type="float" name="max_distance" units="m">Maximum distance the sensor can measure.</field>
     </message>
+    <message id="11038" name="WATER_DEPTH">
+      <description>Water depth</description>
+      <field type="uint32_t" name="time_boot_ms" units="ms">Timestamp (time since system boot)</field>
+      <field type="uint8_t" name="id" instance="true">Onboard ID of the sensor</field>
+      <field type="uint8_t" name="healthy">Sensor data healthy (0=unhealthy, 1=healthy)</field>
+      <field type="int32_t" name="lat" units="degE7">Latitude</field>
+      <field type="int32_t" name="lng" units="degE7">Longitude</field>
+      <field type="float" name="altitude" units="m">Altitude (MSL) of vehicle</field>
+      <field type="float" name="roll" units="rad">Roll angle</field>
+      <field type="float" name="pitch" units="rad">Pitch angle</field>
+      <field type="float" name="yaw" units="rad">Yaw angle</field>
```

Comment:
Do we need to specify the order in which these are applied to get the vehicle pose?

As stated on the call, the fact we've (optionally) replaced `ATTITUDE` with `ATTITUDE_QUATERNION` should be a hint we should use a quaternion IMO.  My brain prefers euler angles but I know that's not the best way to transport the data.

Reply:
We don't seem to specify the axis order in other messages so I think we can leave that out.  I think we decided to stick with euler angles because it is easier for consumers to understand.